sound/soc/sof/intel/apl.c

```diff
@@ -18,13 +18,28 @@
 #include "../sof-priv.h"
 #include "hda.h"
 #include "../sof-audio.h"
+#if IS_ENABLED(CONFIG_SND_SOC_SOF_CLIENT)
+#include "../sof-client.h"
+#endif
 
 static const struct snd_sof_debugfs_map apl_dsp_debugfs[] = {
 	{"hda", HDA_DSP_HDA_BAR, 0, 0x4000, SOF_DEBUGFS_ACCESS_ALWAYS},
 	{"pp", HDA_DSP_PP_BAR,  0, 0x1000, SOF_DEBUGFS_ACCESS_ALWAYS},
 	{"dsp", HDA_DSP_BAR,  0, 0x10000, SOF_DEBUGFS_ACCESS_ALWAYS},
 };
 
+#if IS_ENABLED(CONFIG_SND_SOC_SOF_CLIENT)
+static void apl_register_clients(struct snd_sof_dev *sdev)
+{
+#if IS_ENABLED(CONFIG_SND_SOC_SOF_DEBUG_IPC_FLOOD_TEST_CLIENT)
+	/* this can fail but errors cannot be propagated */
+	sof_client_dev_register(sdev, "sof-ipc-test");
+#endif
+}
+#else
+static void apl_register_clients(struct snd_sof_dev *sdev) {}
```

Comment:
@lyakh no I dont think we expect distros to enable CLIENT support by default. The ask is to support multi-client as an extra feature. If not enabled, the SOF driver would provide basic audio support

Reply:
@lyakh the difference with what you're suggesting is that I'd also have to add the #if to where I set the .register_clients op in all the platforms. It was @plbossart suggestion to have a dummy method here to avoid the switch when setting the op.

Reply:
@ranj063 basic support - exactly. Isn't it the task of distros to provide maximum possible functional support to users? Why would a distro just cripple its users to "basic support" only?
As for `#ifdef` - you only call `.register_clients()` now if `CLIENT` support is enabled. So, you don't need this `#ifdef` for each platform any more - you'll just (potentially) waste a couple of K of RAM for really weird IMHO configurations, that enable SOF but disable `CLIENT` support.